apps/web/src/hooks/useRedirectURL.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback } from 'react';
+import { createCookieHandler } from '../utils/cookies';
+import { assertProtocol } from '../utils/url';
+
+const novuRedirectURLCookie = createCookieHandler('nv_redirect_url');
+
+const REDIRECT_URL_SEARCH_PARAM = 'redirect_url';
+
+export function useRedirectURL() {
+  const setRedirectURL = useCallback(() => {
+    const redirectURLFromParams = new URL(window.location.href).searchParams.get(REDIRECT_URL_SEARCH_PARAM) || '';
+
+    // If there is a redirect URL in the URL, set it in the cookie.
+    if (redirectURLFromParams) {
+      // Protect against XSS attacks via the javascript: pseudo protocol.
+      assertProtocol(redirectURLFromParams);
+      // Expires in 7 days.
+      novuRedirectURLCookie.set(redirectURLFromParams, { expires: 7 });
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** Extract to constant (no magic numbers)‏